File: 1992-find-all-groups-of-farmland/1992-find-all-groups-of-farmland.java
class Solution {
    public int[][] findFarmland(int[][] land) {
        int r = land.length;
        int c = land[0].length;
        
        List<int[]> list = new ArrayList<>();
        
        for(int i = 0; i < r; i++){
            for(int j = 0; j < c; j++){
                if(land[i][j] == 1){
                    
                    int x = i, y = j;
                                        
                    for(x = i; x < r && land[x][j] == 1; x++){
                        for(y = j; y < c && land[x][y] == 1; y++){
                            land[x][y] = 0;
                        }
                    }
                    
                    list.add(new int[]{i, j, x - 1, y - 1});
                }
            }
        }
        
        return list.toArray(new int[list.size()][]);
    }
}

File: 1992-find-all-groups-of-farmland/1992-find-all-groups-of-farmland.cpp
class Solution {
public:
    vector<vector<int>> findFarmland(vector<vector<int>>& land) {
        int r = land.size();
        int c = land[0].size();
        
        vector<vector<int>> res;
        
        for(int i = 0; i < r; i++){
            for(int j = 0; j < c; j++){
                if(land[i][j] == 1){
                    
                    land[i][j] = 0;
                    
                    int ei = i, ej = j;
                    
                    while(ei + 1 < r && land[ei + 1][j] == 1){
                        ++ei;    
                    }
                    
                    while(ej + 1 < c && land[i][ej + 1] == 1){
                        ++ej;    
                    }                   
                    
                    res.push_back({i, j, ei, ej});
                    
                    for(int ti = i; ti <= ei; ti++){
                        for(int tj = j; tj <= ej; tj++){
                            land[ti][tj] = 0;
                        }
                    }
                    
                }
            }
        }
        
        return res;
    }
};